daemon/containerd/image_list.go

```diff
@@ -199,14 +215,30 @@ func (i *ImageService) singlePlatformImage(ctx context.Context, contentStore con
 			repoTags = append(repoTags, rawImg.Name)
 		}
 	} else {
-		repoTags = append(repoTags, reference.TagNameOnly(ref).String())
-
 		digested, err := reference.WithDigest(reference.TrimNamed(ref), target)
 		if err != nil {
 			logger.WithError(err).Error("failed to create digested reference")
 		} else {
 			repoDigests = append(repoDigests, digested.String())
 		}
+
+		for _, referenceDigest := range references {
+			if isDanglingImage(referenceDigest) {
```

Comment:
Haven't fully grasped all the logic yet, but I _think_ filtering by "dangling" is possible, including "only show dangling". Would this line mean that we skip dangling images before checking if there's a "dangling" filter?